pkg/obfuscator/generator.go

```diff
@@ -27,7 +29,22 @@ func (g *generator) generateStaticReplacement() string {
 	return g.static
 }
 
+// generateReplacement returns the replacement based on the replacementType argument
+func (g *generator) generateReplacement(replacementType schema.ObfuscateReplacementType, key string, tracker ReplacementTracker) string {
+	var replacement string
+	switch replacementType {
+	case schema.ObfuscateReplacementTypeStatic:
+		replacement = tracker.GenerateIfAbsent(key, g.generateStaticReplacement)
+	case schema.ObfuscateReplacementTypeConsistent:
+		replacement = tracker.GenerateIfAbsent(key, g.generateConsistentReplacement)
+	}
+	return replacement
+}
+
 // newGenerator creates a generator objects and populates with the provided arguments
-func newGenerator(template, static string) *generator {
-	return &generator{template: template, static: static}
+func newGenerator(template, static string, replacementType schema.ObfuscateReplacementType) (*generator, error) {
+	if replacementType != schema.ObfuscateReplacementTypeStatic && replacementType != schema.ObfuscateReplacementTypeConsistent {
```

Comment:
can you add a test for that statement? Try passing in some other string, I think you need to do a type cast to schema.ObfuscateReplacementType - which is just a string anyway

Reply:
Done.